### PR TITLE
feat: Add Spring WebMVC streamable server transport provider

### PR DESCRIPTION
**Pull request by tzolov**
- Add WebMvcStreamableServerTransportProvider for non-reactive MCP server support
- Implement HTTP/SSE transport with session management and graceful shutdown
- Support GET, POST, DELETE endpoints for MCP protocol operations
- Add test suite including integration tests with Tomcat
- Refactor AbstractMcpAsyncServerTests to support multiple transport implementations
- Provide WebMVC alternative to existing WebFlux streamable transport
    
Related to #72

NOTE: built on top and depending on the #420 

This PR adds Spring WebMVC streamable server transport provider support to the MCP Java SDK, providing a non-reactive alternative to the existing WebFlux implementation for HTTP/SSE-based MCP servers.

## How Has This Been Tested?

The implementation has been thoroughly tested with:

- **Unit tests**: `WebMcpStreamableMcpAsyncServerTests` extending the abstract test suite
- **Integration tests**: `WebMvcStreamableIntegrationTests` with comprehensive scenarios including:

## Breaking Changes

No breaking changes. 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation update

## Checklist

- [x] I have read the [MCP Documentation](https://modelcontextprotocol.io)
- [x] My code follows the repository's style guidelines
- [x] New and existing tests pass locally
- [x] I have added appropriate error handling
- [x] I have added or updated documentation as needed

## Additional context

**Implementation Details:**
- `WebMvcStreamableServerTransportProvider` implements `McpStreamableServerTransportProvider`
- Uses Spring WebMVC's `RouterFunction` for endpoint handling (GET, POST, DELETE)
- Leverages Server-Sent Events (SSE) via `ServerResponse.sse()` for streaming responses
- Maintains session state using `ConcurrentHashMap` for thread safety
- Supports message replay functionality for connection recovery
- Includes builder pattern for easy configuration

**Key Features:**
- Session management with unique session IDs
- Graceful shutdown with proper resource cleanup
- Message replay support for connection resilience
- Configurable endpoint paths and DELETE request handling
- Full compatibility with existing MCP protocol features (tools, resources, prompts, sampling, elicitation)

**Testing Strategy:**
- Refactored `AbstractMcpAsyncServerTests` to use `prepareAsyncServerBuilder()` pattern
- This allows the same test suite to validate both WebFlux and WebMVC implementations
- Added transport-specific integration tests for WebMVC-specific functionality
- Comprehensive error handling and edge case coverage

### CODE REVIEW

File: mcp-spring/mcp-spring-webflux/src/main/java/io/modelcontextprotocol/server/transport/WebFluxStreamableServerTransportProvider.java [hunk 1]
```diff
@@ -0,0 +1,479 @@
...
+	 * connection and sets up the SSE event stream.
+	 * @param request The incoming server request
+	 * @return A Mono which emits a response with the SSE event stream
+	 */
+	private Mono<ServerResponse> handleGet(ServerRequest request) {
+		if (isClosing) {
+			return ServerResponse.status(HttpStatus.SERVICE_UNAVAILABLE).bodyValue("Server is shutting down");
+		}
+
+		McpTransportContext transportContext = this.contextExtractor.apply(request);
```
Comment (tzolov) [inline mcp-spring/mcp-spring-webflux/src/main/java/io/modelcontextprotocol/server/transport/WebFluxStreamableServerTransportProvider.java:195]: Not sure how to use this yet?

File: mcp-spring/mcp-spring-webmvc/src/main/java/io/modelcontextprotocol/server/transport/WebMvcStreamableServerTransportProvider.java [hunk 2]
```diff
@@ -0,0 +1,616 @@
...
+	 * Handles GET requests for SSE connections and message replay.
+	 * @param request The incoming server request
+	 * @return A ServerResponse configured for SSE communication, or an error response
+	 */
+	private ServerResponse handleGet(ServerRequest request) {
+		if (this.isClosing) {
+			return ServerResponse.status(HttpStatus.SERVICE_UNAVAILABLE).body("Server is shutting down");
+		}
+
+		McpTransportContext transportContext = this.contextExtractor.apply(request);
```
Comment (tzolov) [inline mcp-spring/mcp-spring-webmvc/src/main/java/io/modelcontextprotocol/server/transport/WebMvcStreamableServerTransportProvider.java]: How to make use of this?

File: mcp-spring/mcp-spring-webflux/src/main/java/io/modelcontextprotocol/server/transport/WebFluxStreamableServerTransportProvider.java [hunk 3]
```diff
@@ -0,0 +1,479 @@
...
+
+			String sessionId = request.headers().asHttpHeaders().getFirst("mcp-session-id");
+
+			McpStreamableServerSession session = this.sessions.get(sessionId);
+
+			if (session == null) {
+				return ServerResponse.notFound().build();
+			}
+
+			return session.delete().then(ServerResponse.ok().build());
```
Comment (tzolov) [inline mcp-spring/mcp-spring-webflux/src/main/java/io/modelcontextprotocol/server/transport/WebFluxStreamableServerTransportProvider.java]: I guess we need to remove the session from the sessions maps as well? 
e.g. `this.sessions.remove(sessionId);`

**Comment by WeichenYe** [comment]
Hello! Could you share when this change is expected to be included in an official release?
